Server/Components/Devices/DevicesFrame.razor.cs

```diff
@@ -50,24 +54,33 @@ public partial class DevicesFrame : AuthComponentBase, IDisposable
     private IDataService DataService { get; init; } = null!;
 
     [Inject]
-    private IJsInterop JsInterop { get; init; } = null!;
+    private IToastService ToastService { get; init; } = null!;
 
     [Inject]
-    private IToastService ToastService { get; init; } = null!;
+    private IMessenger Messenger { get; init; } = null!;
 
     private int TotalPages => (int)Math.Max(1, Math.Ceiling((decimal)_filteredDevices.Count / _devicesPerPage));
 
     public void Dispose()
     {
-        CircuitConnection.MessageReceived -= CircuitConnection_MessageReceived;
-        AppState.PropertyChanged -= AppState_PropertyChanged;
+        Messenger.Unregister<DisplayNotificationMessage, string>(this, CircuitConnection.ConnectionId);
```

Comment:
I'd need to update the `SimpleMessenger` package and publish a new version to NuGet.org.  Can this one wait?

Reply:
Actually, I think I found a deadlock in the package after the last update.  I'll go ahead and make this change while I'm in there.

Reply:
After testing this, I'm having reservations.  Any kind of disposable I return (e.g. below screenshots) will need a reference to the subscriber.

This can lead to a situation where a reference to the `HandlerRegistration` exists somewhere that is preventing it from getting GCed, which would prevent the subscriber from being GCed, which would prevent the underlying `ConditionalWeakTable` from automatically removing the handler.

I was able to produce this effect in tests easily, but I don't know how likely this is to happen in practice.  You'd need to store the disposable in an outer scope (e.g. cache/store), and forget to remove it, or have something else that still references it.

![image](https://github.com/immense/Remotely/assets/20995508/d74c403c-c0fd-4a02-a851-d0b6e9d695f9)

![image](https://github.com/immense/Remotely/assets/20995508/ffd15b6b-908a-4085-afb9-f0ae844f1e6c)

Reply:
For documentation.  As I said in chat, I suppose it's perfectly reasonable to expect the caller to know not to hold a long-lived reference to a disposable like this.  And I think this would look cleaner.  So I'll continue with it and open a PR in the other repo.